ERCS/erc-7811.md

```diff
@@ -0,0 +1,239 @@
+---
+eip: 7811
+title: Wallet Asset Discovery
+description: JSON-RPC method for wallets to share a user’s full asset list with Dapps, including assets not easily found through on-chain data alone
+author: Luka Isailovic (@lukaisailovic), Konrad Kopp (@kopy-kat), Derek Rein (@arein)
+discussions-to: https://ethereum-magicians.org/t/erc-7811-wallet-asset-discovery/21639
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-11-07
+requires: 20, 155, 721, 5792
+---
+
+## Abstract
+
+This ERC introduces a new RPC call, `wallet_getAssets`, for wallets to declare to the Dapp what assets are owned by the user. This allows for more accurate asset discovery and the use of assets that aren’t available on-chain but can be provided by the wallet
+
+## Motivation
+
+Currently, Dapps primarily rely on on-chain data to determine a user's balance, which can be limiting. Furthermore, a Dapp might restrict the user from initiating actions that the wallet could otherwise resolve, as it cannot account for the total assets a user has across different accounts or chains.
+
+Wallets already have information about a user's assets, including those not visible on-chain, and need a way to communicate that information to Dapps.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Method: `wallet_getAssets`
+
+#### Request schema
+
+```ts
+type Hex = `0x${string}`;
```

Comment:
Suggest defining more type names such as Address and AssetType so it's more clear the semantics in other places.

```suggestion
type Hex = `0x${string}`;
type Address = Hex;
type AssetType = "native" | "erc20" | "erc720" | string;
```